Sources/Proxy/Representations/VolumeRepresentationProxy/index.js

```diff
@@ -101,36 +101,39 @@ function updateDomains(dataset, dataArray, model, updateProp) {
 // ----------------------------------------------------------------------------
 
 function updateConfiguration(dataset, dataArray, { mapper, property }) {
-  const dataRange = dataArray.getRange();
-
   // Configuration
   // actor.getProperty().setInterpolationTypeToFastLinear();
   property.setInterpolationTypeToLinear();
 
-  // For better looking volume rendering
-  // - distance in world coordinates a scalar opacity of 1.0
-  property.setScalarOpacityUnitDistance(
-    0,
-    vtkBoundingBox.getDiagonalLength(dataset.getBounds()) /
-      Math.max(...dataset.getDimensions())
-  );
-  // - control how we emphasize surface boundaries
-  //  => max should be around the average gradient magnitude for the
-  //     volume or maybe average plus one std dev of the gradient magnitude
-  //     (adjusted for spacing, this is a world coordinate gradient, not a
-  //     pixel gradient)
-  //  => max hack: (dataRange[1] - dataRange[0]) * 0.05
-  property.setGradientOpacityMinimumValue(0, 0);
-  property.setGradientOpacityMaximumValue(
-    0,
-    (dataRange[1] - dataRange[0]) * 0.05
-  );
-  // - Use shading based on gradient
-  property.setShade(true);
-  property.setUseGradientOpacity(0, true);
-  // - generic good default
-  property.setGradientOpacityMinimumOpacity(0, 0.0);
-  property.setGradientOpacityMaximumOpacity(0, 1.0);
+  const numberOfComponents = dataArray.getNumberOfComponents();
+  for (let component = 0; component < numberOfComponents; component++) {
+    // For better looking volume rendering
+    // - distance in world coordinates a scalar opacity of 1.0
+    property.setScalarOpacityUnitDistance(
+      component,
+      vtkBoundingBox.getDiagonalLength(dataset.getBounds()) /
+        Math.max(...dataset.getDimensions())
+    );
```

Comment:
Since that is const across components, should you have it computed outside of the loop?

Reply:
Good catch -- done!